tests/test_qp.py

```diff
@@ -139,138 +177,253 @@ def test_create_qp_ex_no_attr_connected(self):
             with PD(ctx) as pd:
                 with CQ(ctx, 100, None, None, 0) as cq:
                     for i in range(1, attr.phys_port_cnt + 1):
-                        qpts = get_qp_types(ctx, i)
-                        for qpt in qpts:
+                        if u.is_eth(ctx, i) and u.is_root():
                             # Extended QP
-                            qia = get_qp_init_attr_ex(cq, pd, attr, attr_ex,
-                                                      qpt)
-                            caps = qia.cap  # Save them to verify values later
+                            qia = get_qp_init_attr_ex(cq, pd, attr, attr_ex, e.IBV_QPT_RAW_PACKET)
+                            caps = qia.cap # Save them to verify values later
                             try:
                                 qp = QP(ctx, qia)
                             except PyverbsRDMAError as ex:
                                 if ex.error_code == errno.EOPNOTSUPP:
-                                    raise unittest.SkipTest('Create QP with extended attrs is not supported')
+                                    raise unittest.SkipTest('Create RAW Packet QP with extended attrs is not supported')
                                 raise ex
-                            qp_attr, qp_init_attr = qp.query(e.IBV_QP_STATE |
-                                                             e.IBV_QP_CAP)
-                            verify_qp_attrs(caps, e.IBV_QPS_RESET, qp_init_attr,
-                                            qp_attr)
+                            qp_attr, qp_init_attr = qp.query(e.IBV_QP_STATE | e.IBV_QP_CAP)
+                            verify_qp_attrs(caps, e.IBV_QPS_RESET, qp_init_attr, qp_attr)
                             # Legacy QP
                             qia = get_qp_init_attr(cq, attr)
-                            qia.qp_type = qpt
-                            caps = qia.cap  # Save them to verify values later
-                            qp = QP(pd, qia)
-                            qp_attr, qp_init_attr = qp.query(e.IBV_QP_STATE |
-                                                             e.IBV_QP_CAP)
-                            verify_qp_attrs(caps, e.IBV_QPS_RESET, qp_init_attr,
-                                            qp_attr)
+                            qia.qp_type = e.IBV_QPT_RAW_PACKET
+                            caps = qia.cap
+                            try:
+                                qp = QP(pd, qia)
+                            except PyverbsRDMAError as ex:
+                                if ex.error_code == errno.EOPNOTSUPP:
+                                    raise unittest.SkipTest('Create RAW Packet QP is not supported')
+                                raise ex
+                            qp_attr, qp_init_attr = qp.query(e.IBV_QP_STATE | e.IBV_QP_CAP)
+                            verify_qp_attrs(caps, e.IBV_QPS_RESET, qp_init_attr, qp_attr)
 
     def test_modify_qp(self):
```

Comment:
can you please change the name to test_modify_ud_qp? (and update the docstring accordingly)

Reply:
Yes, I'll fix it in v2.